helios-client/src/main/java/com/spotify/helios/common/JobValidator.java

```diff
@@ -491,6 +500,37 @@ private boolean validateRepositoryName(final String repositoryName,
     return errors;
   }
 
+  /**
+   * Validate the Job's added Linux capabilities.
+   * @param job The Job to check.
+   * @return A set of error Strings
+   */
+  private Set<String> validateAddedCapabilities(final Job job) {
+    final Set<String> caps = job.getAddedCapabilities();
+
+    if (caps == null) {
+      return emptySet();
+    }
+
+    final Set<String> errors = Sets.newHashSet();
+
+    final Set<String> disallowedCaps = Sets.newHashSet();
+    for (final String cap : caps) {
+      if (!whitelistedCapabilities.contains(cap)) {
+        disallowedCaps.add(cap);
+      }
+    }
```

Comment:
I think this can also be `final Set<String> disallowedCaps = Sets.difference(caps, whitelistedCapabilities)`